Add a vignette on the classification of effects (#3)
inbo/effectclass: vignettes/classification.Rmd

---
title: "Standardised Classification of Effects Based on Their Uncertainty"
author: Thierry Onkelinx
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Standardised Classification of Effects Based on Their Uncertainty}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
library(knitr)
opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
library(ggplot2)
```

## A Simple, Stable and Transparent Classification of Effects

> The `effectclass` package provides a functions to classify effects in a stable and transparent manner.

Reviewer: ```suggestion
The `effectclass` package provides functions to classify effects in a stable and transparent manner.
```

We base the classification on the lower and upper bounds of a confidence (or credible) interval, focussing on the uncertainty rather than the point estimates.
First, we compare the confidence interval with a reference.
This reference typically refers to the null hypothesis.
An effect with confidence interval above the reference is a significant (important) effect.
Let's call this a `positive effect`.
Likewise, a `negative effect` has a confidence interval below the reference.

```{r base-effect, fig.cap = "Comparing confidence intervals to the reference.", echo = FALSE}
ds <- data.frame(
  effect = factor(1:2, labels = c("positive\neffect", "negative\neffect")
  ),
  lcl = c(0.1, -0.5),
  ucl = c(0.5, -0.1)
)
ggplot(ds, aes(ymin = lcl, ymax = ucl, x = effect)) +
  geom_errorbar() +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_y_continuous(
    breaks = 0, 
    labels = "reference"
  ) +
  theme(
    axis.title = element_blank()
  )
```

> But what when the confidence interval contains the reference?

Reviewer: ```suggestion
But what if the confidence interval contains the reference?
```

Then the width of the confidence interval plays an important role.
Suppose the confidence interval is wide.
The value of the effect can range from a strong negative effect over no effect to a strong positive effect.
Basically, we have no information on the effect.
On the other hand, suppose we have a narrow confidence interval.
The effect still is either positive or negative, but we know that it won't be a strong positive nor strong negative effect.
We could define a lower^[below the reference] and upper^[above the reference] threshold and use them to separate confidence intervals that contain the reference.
If the confidence interval is between both the upper and lower threshold, we can state that the effect does not exceed these thresholds.
We set the thresholds at a level where we can consider the effect to be not relevant so we can call this `no effect`.

> As soon as the confidence interval contains the reference and at least on of the threshold, we consider it an `unknown effect`.

Reviewer: ```suggestion
As soon as the confidence interval contains the reference and at least one of the thresholds, we consider it an `unknown effect`.
```

At the same time, the confidence interval is also below the upper threshold which would lead to `no effect` if the confidence interval contains the reference.
We suggest the term `moderate positive effect` for this situation.
A confidence interval above the upper threshold is outside of the `no effect` zone.
We suggest the term `strong positive effect` for this situation.
We keep the term `positive effect` when the confidence interval is above the reference and contains the upper threshold.

```{r detailed-effect, fig.cap = "Comparing confidence intervals to the reference and thresholds for positive effects.", echo = FALSE}
ds <- data.frame(
  effect = factor(
    1:3, 
    labels = c(
      "positive\neffect", "moderate\npositive\neffect", 
      "strong\npositive\neffect"
    )
  ),
  lcl = c(0.1, 0.1, 0.3),
  ucl = c(0.5, 0.2, 0.5)
)
ggplot(ds, aes(ymin = lcl, ymax = ucl, x = effect)) +
  geom_errorbar() +
  geom_hline(yintercept = c(0, -0.25, 0.25), linetype = c(2, 3, 3)) +
  scale_y_continuous(
    breaks = c(-0.25, 0, 0.25), 
    labels = c("lower\nthreshold", "reference", "upper\nthreshold")
  ) +
  theme(
    axis.title = element_blank()
  )
```

> We split the `negative effect` in a `strong negative effect`, `moderate negative effect` and `negative effect` using the similar rules as for the `positve effect`.

Reviewer: ```suggestion
We split the `negative effect` in a `strong negative effect`, `moderate negative effect` and `negative effect` using analogous rules as for the `positve effect`.
```

Reviewer: or remove "the " from "the similar"

  ) +
  theme(
    axis.title = element_blank()
  )
```

## Classification Using `effectclass`

The `effectclass` package provides `classification()` for classifying confidence intervals.
The function requires a vector of lower bounds of the confidence intervals (`ucl`) and a vector of upper bounds (`ucl`).
It also requires the lower and upper thresholds (`threshold`) as a two-element vector.
In case the thresholds are symmetric around the `reference`, you can provide a single value which is the absolute difference between the reference and both thresholds.
The default `reference` is zero.
`classification()` returns a ten level factor using the two digits symbols mentioned above.
The levels range from `strong positive effect` over `no effect` to `strong negative effect`.
The `potential` and `unknown` levels end the row.

```{r}
set.seed(20191126)
n <- 50
ds <- data.frame(
  point_estimate = runif(n, min = -2, max = 2),
  se = runif(n)
)
ds$lcl <- qnorm(p = 0.025, mean = ds$point_estimate, sd = ds$se)
ds$ucl <- qnorm(p = 0.975, mean = ds$point_estimate, sd = ds$se)
library(effectclass)
ds$effect <- classification(lcl = ds$lcl, ucl = ds$ucl, threshold = 1)

> table(ds$effect)

Reviewer: Maybe show `kable(ds)` instead of `table(ds$effect)`? I think this is easier to follow. In that case consider using the code you used on lines 187-192 to demonstrate the functionality, rather than using `runif()` (in order to restrict the number of rows of the table)

Author: Updated in a7e65cc
